### PR TITLE
test(e2e): multizone upgrade

### PR DESCRIPTION
**Pull request by lobkovilya**
Add an e2e test to check multizone upgrades.

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e/helm/kuma_helm_upgrade_multizone.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			n, err := numberOfPolicies(global)
+			g.Expect(err).ToNot(HaveOccurred())
+			return n
+		}, "10s", "100ms").Should(Equal(1))
+		// and
+		Consistently(func(g Gomega) int {
+			n, err := numberOfPolicies(zone)
+			g.Expect(err).ToNot(HaveOccurred())
+			return n
+		}, "10s", "100ms").Should(Equal(1))
```
Comment (jakubdyszkiewicz) [inline test/e2e/helm/kuma_helm_upgrade_multizone.go:152]: nit: can we squeeze this to one consistently? 20s overhead is a lot :( Should this be 5s?

File: pkg/util/versions/versions.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"os"
+
+	"github.com/Masterminds/semver/v3"
+	"sigs.k8s.io/yaml"
+)
+
+const previewVersion = "preview"
+
+func Supported(path string) ([]*semver.Version, error) {
```
Comment (jakubdyszkiewicz) [inline pkg/util/versions/versions.go:12]: Naming is a bit weird here. It's hard to guess what this function does by the name.
It has nothing to do with supported versions, no?

It's more of `ParseFromFile`?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: test/e2e/helm/kuma_helm_upgrade_multizone.go (hunk 1 above)
Comment (lobkovilya) [reply test/e2e/helm/kuma_helm_upgrade_multizone.go:152]: yeah, can be squeezed into 1 Consistently. Unfortunately, 5s is not enough to track the issue, but maybe I can wait for a new subscription on ZoneInsight with Eventually and then do a 5s Consistently that checks the number of policies

File: pkg/util/versions/versions.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/util/versions/versions.go:12]: ah right, some of them are out of support, I'll rename the func

File: pkg/util/versions/versions.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			return nil, err
+		}
+		versions = append(versions, ver)
+	}
+
+	return versions, nil
+}
+
+func OldestUpgradableToLatest(version []*semver.Version) string {
+	return version[len(version)-1-2].String()
```
Comment (michaelbeaumont) [inline pkg/util/versions/versions.go:42]: This isn't quite right I don't think. Shouldn't it be something like `{current.major, current.minor-2, 0}`? This just returns the 3rd to last version in the list which at `2.6.3` could be `2.6.1`

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: pkg/util/versions/versions.go (hunk 3 above)
Comment (michaelbeaumont) [reply pkg/util/versions/versions.go:42]: `versions.yaml` only has the latest for a minor version but https://kuma.io/docs/2.6.x/production/upgrades-tuning/upgrades/#upgrade-kuma explicitly says the patch version doesn't matter, so I suggest setting the patch of the returned version to `0`.

File: test/e2e/helm/kuma_helm_upgrade_multizone.go (hunk 1 above)
Comment (lobkovilya) [reply test/e2e/helm/kuma_helm_upgrade_multizone.go]: Added Eventually that waits for the new ZoneInsight subscription. So now I was able to create 1 Consistently with 5s.

File: pkg/util/versions/versions.go (hunk 3 above)
Comment (michaelbeaumont) [reply pkg/util/versions/versions.go:42]: @lobkovilya let's leave it like you have it and open a website PR to make the support policy explicit.

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**